lib/screens/create_region_screens/interactor/mappers/generate_region_id_state_mapper.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:seeds/domain-shared/page_state.dart';
+import 'package:seeds/domain-shared/result_to_state_mapper.dart';
+import 'package:seeds/screens/create_region_screens/interactor/viewmodels/create_region_bloc.dart';
+
+class GenerateRegionIdStateMapper extends StateMapper {
+  CreateRegionState mapResultToState(CreateRegionState currentState) {
+    //Generate a suggested Region Id for the user based on the Region name they entered
+    String suggestedRegionId = currentState.regionName.toLowerCase().trim().split('').map((character) {
```

Comment:
i dont think we need to split this into an array of chars to validate. 

We can run Regex on strings to assert the string matches our allowed values. 

Where did u copy this logic from?

Reply:
Generate Username